### DISCUSSION TITLE
JSON UQL How to remove duplicates and get totals

### DISCUSSION THREAD

**Discussion by M-Matos** [grafana/grafana-infinity-datasource#469] [Q&A]
Hi at the moment I am getting data in json format, if I use the default parser and and the list path of each of the json item I can list it with no problem. the list that I get contains items that are duplicated. I will like to remove the duplicate items or group them and add a new column with the count of how many times it repeats. 
Thanks 
Example of what I the current  table looks

ZGrab Application Layer Scanner Detection
SIPVicious Scanner Detection
SIPVicious Scanner Detection
Masscan Port Scanning Tool Detection
Masscan Port Scanning Tool Detection
Masscan Port Scanning Tool Detection 
 
Expected Output - Version 1- This will be to show like a top 10 attacks list
```
ZGrab Application Layer Scanner Detection
SIPVicious Scanner Detection
Masscan Port Scanning Tool Detection
```
Expected Output - Version 2

Attack Name                                                         total hits                
ZGrab Application Layer Scanner Detection               1
Masscan Port Scanning Tool Detection                      3
SIPVicious Scanner Detection                                     2

With this output, I can do some nice pie charts.

This is a sample data form the json. 
{
      "created_time": {
        "display_value": "18/10/2022 06:36 AM",
        "value": "1666089370692"
      },
      "udf_fields": {
        "udf_sline_2128": "United States",
        "udf_sline_2122": "SIPVicious Scanner Detection",

      },
      "id": "24143"
    },
    {
      "created_time": {
        "display_value": "19/10/2022 08:04 PM",
        "value": "1666224271374"
      },
      "udf_fields": {
        "udf_sline_2128": "United States",
        "udf_sline_2122": "Masscan Port Scanning Tool Detection",
      },
      "id": "24387"
    },
    {
      "created_time": {
        "display_value": "25/10/2022 03:26 AM",
        "value": "1666682791440"
      },
      "udf_fields": {
        "udf_sline_2128": "United States",
        "udf_sline_2122": "Masscan Port Scanning Tool Detection",
      },
      "id": "25126"
    }

**Comment by yesoreyeram** [comment]
Did you try summarize option in uql?

**Reply by M-Matos** [reply]
Hi yesoreyeram, thanks for the fast responce,  I have been trying but I have no idea how to make it work, do you have some example besides the one on the documentation page or can you give me some suggestions thanks.

**Reply by M-Matos** [reply]
Hi yesoreyeram, I still trying to acomplish this but without luck. can you help me figure this out. Thanks
